source/common/stream_info/utility.cc

```diff
@@ -130,6 +130,14 @@ absl::optional<std::chrono::nanoseconds> TimingUtility::downstreamHandshakeCompl
   return duration(timing.value().get().downstreamHandshakeComplete(), stream_info_);
 }
 
+absl::optional<std::chrono::nanoseconds> TimingUtility::upstreamHandshakeComplete() {
+  OptRef<const UpstreamTiming> timing = stream_info_.upstreamTiming();
```

Comment:
All the info are already available. You can just get the timing : `OptRef<const UpstreamTiming> timing = getUpstreamTiming(stream_info_);`

Reply:
Changed